poetry_dynamic_versioning/__init__.py

```diff
@@ -142,6 +144,26 @@ def alt_poetry_create(cls, *args, **kwargs):
     sys.modules["poetry.poetry"] = poetry_factory_module
 
 
+def _patch_poetry_command_run() -> None:
+    poetry_command_run_module = _state.original_import_func(
+        "poetry.console.commands.run", fromlist=["RunCommand"]
+    )
+    original_poetry_command_run = poetry_command_run_module.RunCommand.handle
+
+    @functools.wraps(original_poetry_command_run)
+    def alt_poetry_command_run(self, *args, **kwargs):
+        # As of version 1.0.0b2, on Linux, the `poetry run` command
+        # uses `os.execvp` function instead of spawning a new process.
+        # This prevents the atexit `deactivte` hook to be invoked.
+        # For this reason, we immediately call `deactivate` before
+        # actually executing the run command.
+        deactivate()
```

Comment:
One side effect of calling `deactivate()` first is that the version won't be available to the command, which could be a problem if you have any custom packaging/document generation/etc that needs it. We should be able to do this to make it available:

```python
result = original_poetry_command_run(self, *args, **kwargs)
deactivate()
return result
```

Which could be tested as:

```
# Make sure the default version is replaced during Poetry:
! poetry run grep 'version = "0.0.999"' pyproject.toml
# Make sure the default version is restored after Poetry:
grep 'version = "0.0.999"' pyproject.toml
```

Reply:
Hi, unfortunately this wouldn't work. On Linux, everything after `original_poetry_command_run(self, *args, **kwargs)` wouldn't be executed because the poetry python process has been replaced by the command being executed via `poetry run`.

Reply:
Oh, huh, you're right. I didn't expect that since Poetry does `return os.execvp(...)`.

Reply:
Yeah I know, that return statement is quite misleading :D